crates/kornia-tensor/src/storage.rs

```diff
@@ -108,40 +124,17 @@ impl<T, A: TensorAllocator> TensorStorage<T, A> {
     }
 }
 
-// TODO: pass the allocator to constructor
-impl<T, A: TensorAllocator> From<Vec<T>> for TensorStorage<T, A>
-where
-    A: Default,
-{
-    /// Creates a new tensor buffer from a vector.
-    fn from(value: Vec<T>) -> Self {
-        // Safety
-        // Vec::as_ptr guaranteed to not be null
-        let ptr = unsafe { NonNull::new_unchecked(value.as_ptr() as *mut T) };
-        let len = value.len() * std::mem::size_of::<T>();
-        // Safety
-        // Vec guaranteed to have a valid layout matching that of `Layout::array`
-        // This is based on `RawVec::current_memory`
-        let layout = unsafe { Layout::array::<T>(value.capacity()).unwrap_unchecked() };
-        std::mem::forget(value);
-
-        Self {
-            ptr,
-            len,
-            layout,
-            alloc: A::default(),
-        }
-    }
-}
 // Safety:
 // TensorStorage is thread safe if the allocator is thread safe.
 unsafe impl<T, A: TensorAllocator> Send for TensorStorage<T, A> {}
 unsafe impl<T, A: TensorAllocator> Sync for TensorStorage<T, A> {}
 
 impl<T, A: TensorAllocator> Drop for TensorStorage<T, A> {
     fn drop(&mut self) {
-        self.alloc
-            .dealloc(self.ptr.as_ptr() as *mut u8, self.layout);
+        if !self.managed_by_external {
```

Comment:
@gbin had to add this flag here to prevent double deallocation as as I couldn't find a way to prevent gstreamer deallocating the memory